packages/blade/src/components/Typography/Text/Text.tsx

```diff
@@ -0,0 +1,97 @@
+import type { ReactElement } from 'react';
+import styled from 'styled-components';
+import type { TextTypes } from '../../../tokens/theme/theme';
+import getPlatform from '../../../utils/getPlatform';
+import type { Theme } from '../../BladeProvider';
+import BaseText from '../BaseText';
+import type { BaseTextProps } from '../BaseText/BaseText';
+
+type TextCommonProps = {
+  type: TextTypes;
+  truncateAfterLines?: number;
+  children?: React.ReactNode;
+};
+
+type TextBodyVariant = TextCommonProps & {
+  variant: 'body';
+  weight: keyof Theme['typography']['fonts']['weight'];
+};
+
+type TextCaptionVariant = TextCommonProps & {
+  variant: 'caption';
+  weight: keyof Pick<Theme['typography']['fonts']['weight'], 'regular'>;
+};
+
+export type TextProps<T> = T extends {
+  variant: infer Variant;
+}
+  ? Variant extends 'caption'
+    ? TextCaptionVariant
+    : Variant extends 'body'
+    ? TextBodyVariant
+    : T
+  : T;
+
+type TextForwardedAs = {
+  forwardedAs?: BaseTextProps['as'];
+};
+
+const getProps = <T extends { variant: 'body' | 'caption' }>({
+  variant,
+  type,
+  weight,
+}: Pick<TextProps<T>, 'type' | 'variant' | 'weight'>): BaseTextProps & TextForwardedAs => {
+  const props: BaseTextProps & TextForwardedAs = {
+    color: `surface.text.${type}.lowContrast`,
+    fontSize: 25,
+    fontWeight: weight,
+    fontStyle: 'normal',
+    lineHeight: 's',
+    fontFamily: 'text',
+    forwardedAs: getPlatform() !== 'react-native' ? 'p' : undefined,
```

Comment:
I mean <Text> can also be span right? or are we not allowing that? What if I want span like inline behaviour?

Reply:
Right now we have capped it to `p` only.